Initial work with i18next

eabedrapo opened the pull request:
* Navigation bar elements and <title> with data-i18n attributes (localized text and tooltip) in all pages.
* Language detection based on browser settings.
* JQuery handlers for Language dropdown menu.
* Initial JSON for English and Spanish (TBD: better translation for "About").

**TBD:**
* i18next documentation in our README.md for new collaborators and future maintenance.
* Add localization attribute to other elements in the site (filters, forms, about text, etc).

ekonotek committed:
i18n start

eabedrapo committed:
initial work with i18next to replace jquery.i18n

eabedrapo committed:
initial work with i18next to replace jquery.i18n

eabedrapo committed:
Merge branch 'master' of https://github.com/eabedrapo/migra-map

eabedrapo committed:
Merge branch 'master' of https://github.com/eabedrapo/migra-map

eabedrapo committed:
Merge branch 'master' of https://github.com/eabedrapo/migra-map

yjkogan commented on js/i18napp.js:
```diff
@@ -0,0 +1,27 @@
+  $('.dropdown-menu > li.active').removeClass('active');
+  $(this).addClass('active');
+  $('.dropdown-toggle').text(($(this).text()) + " \u25BC");
+  var language = $(this).children().data('i18n');
```
I think a couple comments in this file about why you're changing which element has the `active` class would help a future maintainer understand this code more clearly. It's hard to tell why the change is necessary from the code alone.

yjkogan reviewed (approved):
Other than that, looks great!

eabedrapo commented:
Hi @yjkogan, thanks for the feedback! You were right. So I added many comments to the JS to make clear what the code is doing so it will be easier to maintain it in the future.

I also realized that I don't need to set the "active" class for the dropdown in the HTML because fallback language is English and that class was applied to the corresponding option after localization anyways. So I removed it from HTML and deleted the JQuery line that was removing it programmatically.

Thanks again for the feedback.
